src/ctelnet.cpp

```diff
@@ -2028,3 +2156,57 @@ void cTelnet::setKeepAlive(int socketHandle)
     setsockopt(socketHandle, IPPROTO_TCP, TCP_KEEPCNT, &count, sizeof(count));
 #endif // defined(Q_OS_WIN32)
 }
+
+// Used to convert a collection of Bytes in the current MUD Server encoding
+// to the UTF-16BE encoding used for QString and then back to a series of bytes
+// as a QByteArray - note that it does NOT retain states between calls as it
+// assumes each call is a complete separate chunk of text - should this not
+// prove to be the case in practice it will be necessary to fork-off seperate
+// instances of this method for each OOB protocol that uses this DECODER:
+QByteArray cTelnet::decodeBytes(const char* bytes)
```

Comment:
If it does not retain state, can this be a const method then?

Reply:
It may call `encodingChanged(...)` which will change a non-static member in the class - does that make it a non-const action.

When I say "it does not retain state" I mean that I am assuming that each time it is called `bytes` represents a complete message and is not a partial message that might happen if the text is split over more than one network packet or if another Telnet protocol with a higher *priority*(?) causes an *interruption* to a particular one.  The bottom line being that there is not a need to hang onto one or more bytes of an incomplete message or symbol in that message (unlike the multi-byte decoding of the UTF-8/GBK/GB18030/BIG5 encoding systems used for the main text data stream which DOES have to handle packet fragmentation of data)...